pkg/ruler/ruler.go

```diff
@@ -1189,6 +1300,8 @@ func (r *Ruler) getShardedRules(ctx context.Context, userID string, rulesRequest
 			return errors.Wrapf(err, "unable to get client for ruler %s", addr)
 		}
 
+		ctx, cancel := context.WithTimeout(ctx, r.cfg.ListRulesFanoutTimeout)
```

Comment:
I think this change is not related to this PR?

Can we remove this from here and create another one where we can discuss possible solutions for this??

Reply:
This is related as we discussed offline. Changed it to `remote_timeout` under ruler_client section